### ISSUE TITLE
Deadlock while opening a document

### ISSUE THREAD

**Issue by tmat** [dotnet/project-system#4798]
**Visual Studio Version**:
16.1.28829.227

Memory dump:
%internal_share%\public\tomat\Bugs\PS-4798

```
Microsoft.VisualStudio.Threading.dll!Microsoft.VisualStudio.Threading.AsyncReaderWriterLock.IsLockHeld(Microsoft.VisualStudio.Threading.AsyncReaderWriterLock.LockKind kind, Microsoft.VisualStudio.Threading.AsyncReaderWriterLock.Awaiter awaiter, bool checkSyncContextCompatibility, bool allowNonLockSupportingContext)	Unknown
 	Microsoft.VisualStudio.ProjectSystem.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.ProjectLockService.Microsoft.VisualStudio.ProjectSystem.IProjectLockService.get_IsPassiveWriteLockHeld()	Unknown
 	Microsoft.VisualStudio.ProjectSystem.VS.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectConfig.GetConfigurationGeneralPropertiesAsync(bool preferSnapshot, bool requiredToMatchProjectVersion)	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder<System.__Canon>.Start<Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectConfig.<GetConfigurationGeneralPropertiesAsync>d__83>(ref Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectConfig.<GetConfigurationGeneralPropertiesAsync>d__83 stateMachine) Line 471	C#
 	Microsoft.VisualStudio.ProjectSystem.VS.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectConfig.GetConfigurationGeneralPropertiesAsync(bool preferSnapshot, bool requiredToMatchProjectVersion)	Unknown
 	Microsoft.VisualStudio.ProjectSystem.VS.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode.GetGuidProperty.AnonymousMethod__1()	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start<Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode.(ref Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode.<>c__DisplayClass675_0.<<GetGuidProperty>b__1>d stateMachine) Line 316	C#
 	Microsoft.VisualStudio.ProjectSystem.VS.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode.GetGuidProperty.AnonymousMethod__1()	Unknown
 	Microsoft.VisualStudio.Threading.dll!Microsoft.VisualStudio.Threading.JoinableTaskFactory.ExecuteJob<Microsoft.VisualStudio.Threading.EmptyStruct>(System.Func<System.Threading.Tasks.Task> asyncMethod, Microsoft.VisualStudio.Threading.JoinableTask job)	Unknown
 	Microsoft.VisualStudio.Threading.dll!Microsoft.VisualStudio.Threading.JoinableTaskFactory.RunAsync(System.Func<System.Threading.Tasks.Task> asyncMethod, bool synchronouslyBlocking, Microsoft.VisualStudio.Threading.JoinableTaskCreationOptions creationOptions, System.Delegate entrypointOverride)	Unknown
 	Microsoft.VisualStudio.ProjectSystem.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.ProjectMultiThreadedService.ExecuteSynchronously(System.Func<System.Threading.Tasks.Task> asyncAction)	Unknown
 	Microsoft.VisualStudio.ProjectSystem.VS.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode.GetGuidProperty.AnonymousMethod__0()	Unknown
 	Microsoft.VisualStudio.ProjectSystem.VS.dll!Microsoft.VisualStudio.ProjectSystem.VS.HResult.Invoke(System.Func<Microsoft.VisualStudio.ProjectSystem.VS.HResult> action, System.IServiceProvider vsShellServiceProvider, Microsoft.VisualStudio.ProjectSystem.IProjectFaultHandlerService projectFaultHandlerService, Microsoft.VisualStudio.ProjectSystem.UnconfiguredProject project)	Unknown
 	Microsoft.VisualStudio.ProjectSystem.VS.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode.GetGuidProperty(uint itemId, int propid, out System.Guid guidParam)	Unknown
 	Microsoft.VisualStudio.ProjectSystem.VS.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNodeWrapper.GetGuidProperty(uint itemid, int propid, out System.Guid pguid)	Unknown
 	[Native to Managed Transition]	
 	[Managed to Native Transition]	
 	Microsoft.VisualStudio.Platform.WindowManagement.dll!Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.InvalidateSelectionContext()	Unknown
 	Microsoft.VisualStudio.Platform.WindowManagement.dll!Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.ShowInternal(Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.ShowFlags showFlags)	Unknown
 	Microsoft.VisualStudio.Platform.WindowManagement.dll!Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.Show.AnonymousMethod__507_0()	Unknown
 	Microsoft.VisualStudio.Shell.15.0.dll!Microsoft.VisualStudio.ErrorHandler.CallWithCOMConvention(System.Func<int> method, bool reportError, bool setShellErrorInfo)	Unknown
 	Microsoft.VisualStudio.Platform.WindowManagement.dll!Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.Show()	Unknown
 	Microsoft.VisualStudio.Platform.WindowManagement.dll!Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.MarshalingWindowFrame.Microsoft.VisualStudio.Shell.Interop.IVsWindowFrame.Show.AnonymousMethod__12_0()	Unknown
 	Microsoft.VisualStudio.Shell.15.0.dll!Microsoft.VisualStudio.Shell.ThreadHelper.Invoke<int>(System.Func<int> method)	Unknown
 	Microsoft.VisualStudio.Platform.WindowManagement.dll!Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.MarshalingWindowFrame.Microsoft.VisualStudio.Shell.Interop.IVsWindowFrame.Show()	Unknown
>	Microsoft.VisualStudio.LanguageServices.dll!Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioWorkspaceImpl.OpenDocumentCore(Microsoft.CodeAnalysis.DocumentId documentId, bool activate) Line 986	C#
```

**Comment by drewnoakes** [comment]
Migrated to https://dev.azure.com/devdiv/DevDiv/_workitems/edit/896925

**Comment by davkean** [comment]
This has been fixed and merged into 16.2.
